app/src/test/java/protect/card_locker/DatabaseTest.java

```diff
@@ -399,18 +399,19 @@ public void updateGroupDuplicate() {
         assertEquals(2, DBHelper.getGroupCount(mDatabase));
 
         // Should fail when trying to rename group two to one
-        boolean result3 = DBHelper.updateGroup(mDatabase, "group two", "group one");
+        Group group2_init = DBHelper.getGroupByName(mDatabase, "group two");
+        boolean result3 = DBHelper.updateGroup(mDatabase, group2_init._id, "group one");
         assertFalse(result3);
         assertEquals(2, DBHelper.getGroupCount(mDatabase));
 
         // Rename failed so both should still be the same
-        Group group = DBHelper.getGroup(mDatabase, "group one");
+        Group group = DBHelper.getGroupByName(mDatabase, "group one");
```

Comment:
We should also test if we can get a group by ID, as that's the actual canonical thing.